plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java

```diff
@@ -218,8 +224,14 @@ else if (javaType == double.class) {
             else if (javaType == Int128.class) {
                 DecimalType decimalType = (DecimalType) type;
                 verify(!decimalType.isShort(), "The type should be long decimal");
-                BigDecimal decimal = ((Decimal128) value).bigDecimalValue();
-                type.writeObject(output, Decimals.encodeScaledValue(decimal, decimalType.getScale()));
+                Decimal128 decimal = (Decimal128) value;
+                if (decimal.compareTo(Decimal128.NEGATIVE_ZERO) == 0) {
+                    type.writeObject(output, Decimals.encodeScaledValue(BigDecimal.ZERO, decimalType.getScale()));
+                }
+                else {
+                    BigDecimal result = decimal.bigDecimalValue();
+                    type.writeObject(output, Decimals.encodeScaledValue(result, decimalType.getScale()));
+                }
```

Comment:
I didn't touch `guessFieldType` method because the precision and scale are still unclear in the case of negative zero.